common/system.h

```diff
@@ -386,6 +405,24 @@ class OSystem : Common::NonCopyable {
 		 */
 		kFeatureCursorPalette,
 
+		/**
+		 * Backends supporting this feature allow specifying a mask for a
+		 * cursor instead of a key color.
+		 */
+		kFeatureCursorMask,
+
+		/**
+		 * Backends supporting this feature allow cursor masks to use mode kCursorMaskInvert in mask values,
+		 * which inverts the destination pixel.
+		 */
+		kFeatureCursorMaskInvert,
```

Comment:
These two values seem to me a bit too granular. Is the engine code expected to feed backends with different cursors based on these inverted mask feature flags?

Reply:
I've updated this entry to be a bit more detailed, unfortunately it's even more complicated now, but it is more correct.

Yes, the intent is engine code is supposed to feed different cursors if invert isn't supported because the engine (or cursor loader) will need to decide what color to use for invert pixels if the invert mode that it wants to use isn't supported.

The color invert is because of the complicated Classic MacOS invert semantics.  Both Mac and Win, in theory, were designed with blend behavior `(Destination AND Mask) XOR Cursor` semantics, however in the MacOS default palette, the 0 color is white and the 255 color (or 1 color in BW mode) is black, whereas on Windows it is the opposite.  This created a problem because when true color modes were added, the cursor color that is supposed to invert now has a color value of 0, so MacOS in true color modes uses `(Destination AND Mask) XNOR Color` (which is same as `(Destination AND Mask) XOR (NOT Color)`), so should some game use that, it is up to the backend to figure out how to render it.

Windows cursors do not support this, in theory the data allows it but the WDDM semantics are that any color with AND mask 1 is only either transparent or inverted, and it is only inverted if the color is white.